xwiki-platform-core/xwiki-platform-observation/xwiki-platform-observation-remote/src/main/java/org/xwiki/observation/remote/internal/DefaultRemoteObservationManagerConfiguration.java

```diff
@@ -69,4 +82,45 @@ public String getNetworkAdapter()
     {
         return this.configurationSource.getProperty("observation.remote.networkadapter", "jgroups");
     }
+
+    @Override
+    public String getId()
+    {
+        return this.id;
+    }
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        Path observation = getObservationDirectory();
+        observation.toFile().mkdir();
+        Path idFile = getIdFile();
+        if (!idFile.toFile().exists()) {
```

Comment:
You don't need to go through the old File API to test that a Path exists :) I think its `Files.exists(idFile)`.